aiida/cmdline/commands/cmd_computer.py

```diff
@@ -276,72 +275,215 @@ def setup_computer(non_interactive, **kwargs):
+    help="The absolute path of the directory on the computer where AiiDA will "
+    "run the calculations (typically, the scratch of the computer). You "
+    "can use the {username} replacement, that will be replaced by your "
+    "username on the remote computer")
+@click.option(
+    '-m',
+    '--mpirun-command',
+    prompt="Mpirun command",
+    default="mpirun -np {tot_num_mpiprocs}",
+    cls=InteractiveOption,
+    contextual_default=partial(get_parameter_default, 'mpirun_command'),
+    help="The mpirun command needed on the cluster to run parallel MPI "
+    "programs. You can use the {tot_num_mpiprocs} replacement, that will be "
+    "replaced by the total number of cpus, or the other scheduler-dependent "
+    "replacement fields (see the scheduler docs for more information)",
+    type=MpirunCommandParamType())
+@click.option(
+    '--mpiprocs-per-machine',
+    prompt="Default number of CPUs per machine",
+    cls=InteractiveOption,
+    help="Enter here the default number of MPI processes per machine (node) that "
+    "should be used if nothing is otherwise specified. Pass the digit 0 "
+    "if you do not want to provide a default value.",
+    contextual_default=partial(get_parameter_default, 'mpiprocs_per_machine'),
+    prompt_fn=should_call_default_mpiprocs_per_machine,
+    required_fn=False,
+    type=click.INT,
+)  # Note: this can still be passed from the command line in non-interactive mode
+@options.PREPEND_TEXT()
+@options.APPEND_TEXT()
+@options.NON_INTERACTIVE()
+@click.pass_context
+@with_dbenv()
+def computer_duplicate(ctx, computer, non_interactive, **kwargs):
+    """Duplicate a Computer."""
+    if kwargs['label'] in get_computer_names():
+        echo.echo_critical('A computer called {} already exists'.format(kwargs['label']))
+
+    if not non_interactive:
+        pre, post = ensure_scripts(kwargs.pop('prepend_text', ''), kwargs.pop('append_text', ''), kwargs)
+        kwargs['prepend_text'] = pre
+        kwargs['append_text'] = post
+
+    kwargs['transport'] = kwargs['transport'].name
+    kwargs['scheduler'] = kwargs['scheduler'].name
+
+    computer_builder = ctx.computer_builder
+    for key, value in kwargs.items():
+        if value is not None:
+            setattr(computer_builder, key, value)
+
+    try:
+        computer = computer_builder.new()
+    except (ComputerBuilder.ComputerValidationError, ValidationError) as e:
+        echo.echo_critical('{}: {}'.format(type(e).__name__, e))
+
+    try:
+        computer.store()
+    except ValidationError as err:
+        echo.echo_critical('unable to store the computer: {}. Exiting...'.format(err))
+
+    echo.echo_success('computer "{}" stored in DB.'.format(computer.name))
+    echo.echo_info('pk: {}, uuid: {}'.format(computer.pk, computer.uuid))
+
+    echo.echo_info("Note: before using it with AiiDA, configure it using the command")
+    echo.echo_info("  verdi computer configure {} {}".format(computer.get_transport_type(), computer.name))
```

Comment:
In interactive mode, we should simply prompt the user should for this (because it almost always needs to happen).

Reply:
(Meaning: Configure computer ... now? [Y/n] )

Reply:
Do you now how this would work? I know I can do something like `ctx.invoke(click_command, **kwargs)` to invoke the command, but if I don't pass anything in kwargs, it complains that `Computer` is `None` in the configure command. If I *do* pass it, I get the exception `TypeError: transport_configure_command() got multiple values for keyword argument 'computer'`. Is this complicated because the configure commands are created dynamically?

Reply:
Fixed